### PR TITLE
Add find method to shopcart

### PR DESCRIPTION
**Pull request by thoughts1053**
Adds `find` static method to `Shopcart`. Returns `shopcart` instance if found, otherwise `None`.

All tests passing and keeps 100% code coverage.

### CODE REVIEW

File: models/shopcart.py [hunk 1]
```diff
@@ -39,3 +39,12 @@ def all():
...
         Shopcart.__data = []
+    
+    @staticmethod
+    def find(uid):
+        """ Find a Shopcart by it's uid"""
+        for cart in Shopcart.__data:
+            if cart.uid == uid:
+                return cart
+
+        return None
```
Comment (radhikamattoo) [inline models/shopcart.py]: Should we maybe create a new Error class, like `NotFoundError`, and raise that instead of returning None?

Comment (carpetri) [reply models/shopcart.py]: We only need an error if we to modifying a shopcart that does not exist so an error is not necessary in this case in my opinion.

Comment (thoughts1053) [reply models/shopcart.py]: I think the API should handle 404 not found in case they call a find on a non-existant cart.

**Comment by thoughts1053** [comment]
@carpetri the test is saving 2 carts separately. Then it uses the static find method to find the cart with the uid 5. It then verifies it has the right data (which at that point is just the uid and an empty products dict/list.

**Pull request merged**